### ISSUE TITLE
Missing `cwd` in the launch.json results in `localRoot` nesting itself

### ISSUE THREAD

**Issue by rwxdash** [microsoft/vscode-js-debug#1140]
**Describe the bug**

Let me quickly talk about our use case and then I'll describe the bug. I don't know if this issue is being discussed before or in fact, this is the expected behaviour. Although I did some searching, I apologise if that's the case. Nevertheless, it's better to learn and understand the case from your point of view.

We, as [Thundra](https://thundra.io), have a [VSCode Plugin](https://marketplace.visualstudio.com/items?itemName=thundra.thundra-debugger) to let our users debug their AWS Lambda Functions on the remote.

In this plugin, we also allow our users to change their `localRoot` as they might have different project structures, etc. So, previous to this js-debugger, things were working fine. Today, while testing some other stuff, I realized this bug with the current js-debugger.

Our launch configuration for Node runtime was like the following and we didn't have `cwd` set.

```yaml
node: {
    # ...
    localRoot: "${workspaceFolder}",
    remoteRoot: "/var/task/",
    # ...
},
```

With this configuration; if a user wants to change their `localRoot` to relative `./dist`, the new `localRoot` would resolve into `${workspaceFolder}/dist/dist`, where (new) `localRoot` uses itself to replace `${workspaceFolder}`. 

After some time of debugging, we've found the following lead;

In the case of missing a `cwd` in the launch configuration, js-debugger sets `cwd` to `localRoot`. See;
https://github.com/microsoft/vscode-js-debug/blob/a0349dc533236dbe2c2a913b355fded55cd27a3e/src/ui/configuration/nodeDebugConfigurationResolver.ts#L112-L117

So, in our scenario, the new `localRoot` our users wants to set is `${workspaceFolder}/dist`. This then gets passed into `resolveVariableInConfig` method where it evaluates the variables. See;
https://github.com/microsoft/vscode-js-debug/blob/1c141189c6e71111b5514c990b5db32e6f79ba87/src/configuration.ts#L1106-L1138

In this method, it uses `${workspaceFolder}/dist` to replace `${workspaceFolder}`. Returning from this with `${workspaceFolder}/dist/dist`, the debugger stops and shows the following error as expected; 

```
The configured `cwd` ${workspaceFolder}/dist/dist does not exist.
```

**To Reproduce**

The following snipped should give the `localRoot` output as I've described above. However, it can also be reproduced by trying our debugger at v1.1.8. 

```js
let varName = 'workspaceFolder';
let varValue = '${workspaceFolder}/dist';
let config = varValue;

let out = config.replace(new RegExp(`\\$\\{${varName}\\}`, 'g'), () => {
  if (!varValue) {
    throw new Error(
      `Unable to resolve \${${varName}} in configuration (${JSON.stringify(varName)})`,
    );
  }
  return varValue;
});

console.log(out);
```

**VS Code Version:** 

Version: 1.62.2
Commit: 3a6960b964327f0e3882ce18fcebd07ed191b316
Date: 2021-11-11T20:59:05.913Z (6 days ago)
Electron: 13.5.2
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 20.6.0

**Additional context**

The fix for this on our side is quite trivial. Once we included `cwd: "${workspaceFolder}"` to our launch configuration, it worked just fine. However, in case someone else comes across the same issue or perhaps it's unexpected behaviour, I wanted to open an issue.

My immediate approach would be to set the `cwd` to `${workspaceFolder}` by default in the `resolveDebugConfigurationAsync` method, but I'm not certain of the side-effects.

Would love to hear your comments. 

Bests.

**Comment by connor4312** [comment]
What does the actual `startDebugging` request look like from your extension?

**Comment by rwxdash** [comment]
On `start`, we call the following line:

```js
vscode.debug.startDebugging(undefined, ideDebuggerConf);
```

and, at this stage `ideDebuggerConf` is the following:

```js
{
    address: 'localhost'
    localRoot: '${workspaceFolder}'
    name: 'Thundra Debugger Session (default)'
    port: 54553
    protocol: 'inspector'
    remoteRoot: '/var/task/'
    request: 'attach'
    thundraSession: true
    timeout: 60000
    type: 'node'
}
```

After this, we establish a WebSocket connection to our broker and do the communication over that. So, when a user calls `Change local root` command before `Start debugger`, the value in the `ideDebuggerConf` changes to the given input.

Does that make sense to you?

**Comment by rwxdash** [comment]
Just FYI that I've released a fix for our plugin at version `1.1.10`, missed the `1.1.9` by mistake. So, if you're trying to reproduce, make sure to install `1.1.8`.
